src/activities/pages/GroupActivities.vue

```diff
@@ -200,30 +208,45 @@ export default {
       return new Date(Math.floor(new Date().getTime() / roundTo) * roundTo)
     }
 
+    const queryClient = useQueryClient()
+
+    /*
+      TODO: something funny happening where it sort of freezes when there is query cache...
```

Comment:
Would be interesting to figure out if this is happens during rendering or somewhere in the data management layer...

Reply:
Yeah, indeed. My current iteration of the code is to have it not do any caching for activities and only use backend filters. Which is fast locally, and waiting until that backend one is merged to try it out with more data...

I did try making it only keep the first page cached, but didn't solve that freezing...